gapic/src/main/com/google/gapid/models/ApiContext.java

```diff
@@ -204,6 +211,20 @@ public boolean contains(long index) {
       return Ranges.contains(context.getRangesList(), index) >= 0;
     }
 
+    public CommandRange findClosest(CommandRange range) {
+      int index = Ranges.contains(context.getRangesList(), last(range));
+      if (index >= 0) {
+        return commands(last(range), 1);
+      } else if (index == -1) {
+        // The given range ends before any of our commands.
+        return (context.getRangesCount() == 0) ? null : commands(first(context.getRanges(0)), 1);
+      } else {
```

Comment:
Ah, I quickly parsed the [`contains()`](https://github.com/pmuetschard/gapid/blob/98b422d06a9163cd65a06258012e92f6d6456fd6/gapic/src/main/com/google/gapid/util/Ranges.java#L94-L98) function and my brain freaked out with the double ternary in the lambda that's used as a parameter. Candidate for a function comment perhaps? :smile: